indigo_social/views.py

```diff
@@ -292,3 +293,17 @@ class UserPopupView(AbstractAuthedIndigoView, DetailView):
     slug_url_kwarg = 'username'
     template_name = 'indigo_social/user_popup.html'
     queryset = User.objects
+
+
+class UserProfilePhotoView(View):
+    def get(self, request, **kwargs):
+        username = kwargs['username']
+        nonce = kwargs['nonce']
+
+        user = User.objects.filter(username=username).first()
```

Comment:
Use `get_object_or_404` otherwise you'll get a 500 internal error if the user doesn't exist. Once you have a user, you can use `user.user_profile`.